src/eduid_userdb/actions/chpass/user.py

```diff
@@ -0,0 +1,126 @@
+    """
+    Subclass of eduid_userdb.User with
+    the eduid-actions plugin for password change specific data.
+
+    :param userid: user id
+    :type userid: bson.ObjectId
+    :param passwords: Password list
+    :type passwords: list
+    :param data: userid and password list
+    :type data: dict
+    :param raise_on_unknown: whether to raise an exception if
+                             there is unknown data in the data dict
+    :type raise_on_unknown: bool
+    """
+
+    def __init__(self, userid = None, passwords = None, data = None,
+                                         raise_on_unknown = True):
+        """
+        """
+        if data is None:
+            data = {'_id': userid, 'passwords': passwords}
+
+        if '_id' not in data or data['_id'] is None:
+            raise UserMissingData('Attempting to record passwords '
+                                  'for an unidentified user.')
+        if 'passwords' not in data or data['passwords'] is None:
+            raise UserMissingData('Attempting to record '
+                                  'an unknown password for '
+                                  'the user with id ' + str(data['_id']))
+
+        self._data_in = deepcopy(data)
+        self._data = dict()
+
+        # things without setters
+        _id = self._data_in.pop('_id', None)
+        if not isinstance(_id, bson.ObjectId):
+            _id = bson.ObjectId(_id)
+        self._data['_id'] = _id
+
+        self._passwords = PasswordList(self._data_in.pop('passwords', []))
+
+        self.modified_ts = self._data_in.pop('modified_ts', None)
+
+        if len(self._data_in) > 0:
+            if raise_on_unknown:
+                raise UserHasUnknownData('User {!s} unknown data: {!r}'.format(
+                    self.user_id, self._data_in.keys()
+                ))
+            # Just keep everything that is left as-is
+            self._data.update(self._data_in)
+    
+    @classmethod
+    def from_central_user(cls, user):
+        '''
+        Create user from generic user.
+
+        :param user: user from central db
+        :type user: eduid_userdb.user.User
+        '''
+        data = {
+                '_id': user.user_id,
+                'passwords': user.passwords.to_list_of_dicts(),
+                'modified_ts': user.modified_ts
+            }
+        return cls(data=data)
+
+    # -----------------------------------------------------------------
+    def to_dict(self, old_userdb_format=False):
+        """
+        Return user data serialized into a dict that can be stored in MongoDB.
+
+        :param old_userdb_format: Set to True to get the dict in the old database format.
+        :type old_userdb_format: bool
+
+        :return: User as dict
+        :rtype: dict
+        """
+        res = deepcopy(self._data)  # avoid caller messing up our private _data
+        res['passwords'] = self.passwords.to_list_of_dicts(
+                old_userdb_format=old_userdb_format)
```

Comment:
If I remember right, it was intentional. I think ChpassUser is only ever used by the change_password plugin to save the new password just before updating  the central db, at [1], so it does not need it. However I don't think it will do any harm adding it.

[1] https://github.com/SUNET/eduid_action.change_passwd/blob/eperez-chpasswd/src/eduid_action/change_passwd/action.py#L153